examples/planning/generate_keyword_ideas.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This example generates keyword ideas from a list of seed keywords."""
+
+from __future__ import absolute_import
+
+import argparse
+import six
+import sys
+from google.ads.google_ads.client import GoogleAdsClient
+from google.ads.google_ads.errors import GoogleAdsException
+
+_DEFAULT_LOCATION_IDS = '1023191'  # location ID for New York, NY
+_DEFAULT_LANGUAGE_ID = '1000'  # language ID for English
+
+
+def main(client, customer_id, location_ids, language_id, keywords, page_url):
+    keyword_plan_idea_service = client.get_service('KeywordPlanIdeaService',
+                                                   version='v1')
+    keyword_competition_level_enum = (
+        client.get_type('KeywordPlanCompetitionLevelEnum', version='v1')
+        .KeywordPlanCompetitionLevel)
+    keyword_plan_network = client.get_type(
+        'KeywordPlanNetworkEnum', version='v1').GOOGLE_SEARCH_AND_PARTNERS
+    locations = map_locations_to_string_values(client, location_ids)
+    language = map_language_to_string_value(client, language_id)
+
+    # Only one of these values will be passed to the KeywordPlanIdeaService
+    # depending on whether keywords, a page_url or both were given.
+    url_seed = None
+    keyword_seed = None
+    keyword_url_seed = None
+
+    # Either keywords or a page_url are required to generate keyword ideas
+    # so this raises an error if neither are provided.
+    if (not (len(keywords) or page_url)):
+        raise ValueError('At least one of keywords or page URL is required, '
+                         'but neither was specified.')
+
+    # To generate keyword ideas with only a page_url and no keywords we need
+    # to initialize a UrlSeed object with the page_url as the "url" field.
+    if (not len(keywords) and page_url):
```

Comment:
I recommend an extra set of parens to remove ambiguity, ie. if ((not len(keywords)) and page_url):